### ISSUE TITLE
container user 10001 is part of root group?

### ISSUE THREAD

**Issue by 0x6f677548** [open-telemetry/opentelemetry-collector-contrib#35179]
### Component(s)

cmd/otelcontribcol

### Describe the issue you're reporting

[dockerfile](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/616d91aa870ecc5ed7b1b26fecbd69af6b22ab53/cmd/otelcontribcol/Dockerfile#L9)

It seems to me that the container is running with a user 10001 part of the 0 (root) group. 

I suggest something like:
```
ARG USER_UID=10001
ARG USER_GID=10001
USER ${USER_UID}:${USER_GID}

```
Am I missing something?

**Comment by rogercoll** [comment]
Nice catch! Do you know how we could verify the GID of the user from that image? (Being based on scratch makes debugging hard)

But based on [dockerfile documentation](https://docs.docker.com/reference/dockerfile/#user):

> When the user doesn't have a primary group then the image (or the next instructions) will be run with the root group.

And if we want to follow [security best-practices](https://docs.docker.com/build/building/best-practices/):

> Consider an explicit UID/GID.
Users and groups in an image are assigned a non-deterministic UID/GID in that the "next" UID/GID is assigned regardless of image rebuilds. So, if it’s critical, you should assign an explicit UID/GID.


We should also bring your suggestion to the repository used to publicly release the images https://github.com/open-telemetry/opentelemetry-collector-releases

**Comment by 0x6f677548** [comment]
Yes, it is running as group 0. Just confirmed by running id inside the container shell:```

~ $ id
uid=10001 gid=0 groups=0
```

One could use an alpine image to debug it, but if you want to confirm it on a scratch, here's my suggestion:

```
FROM alpine:latest AS builder

# Install a statically linked shell and the necessary binaries
RUN apk add --no-cache busybox-static
RUN apk add --no-cache coreutils acl attr

FROM otel/opentelemetry-collector-contrib:latest AS prep

FROM scratch

# Copy the shell executable from the builder stage
COPY --from=builder /bin/busybox.static /bin/sh
# Copy id binary from the builder stage
COPY --from=builder /usr/bin/id /bin/id

# Copy required shared libraries
COPY --from=builder /lib/ld-musl-x86_64.so.1 /lib/ld-musl-x86_64.so.1
COPY --from=builder /lib/libc.musl-x86_64.so.1 /lib/libc.musl-x86_64.so.1
COPY --from=builder /lib/libcrypto.so.3 /lib/
COPY --from=builder /lib/libacl.so.1 /lib/
COPY --from=builder /lib/libattr.so.1 /lib/
COPY --from=builder /lib/libutmps.so.0.1 /lib/
COPY --from=builder /lib/libskarnet.so.2.14 /lib/
COPY --from=builder /lib/libutmps.so.0.1 /lib/

COPY --from=prep /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
COPY --from=prep otelcol-contrib /otelcol-contrib

ARG USER_UID=10001
USER ${USER_UID}



# copy the config file to the /etc folder
COPY otel_collector.config.yaml /etc/otel_collector.config.yaml

EXPOSE 4317 55680 55679
ENTRYPOINT ["/otelcol-contrib"]
CMD ["--config", "/etc/otel_collector.config.yaml"]
```
Let me know if I can help somehow.

**Comment by 0x6f677548** [comment]
> We should also bring your suggestion to the repository used to publicly release the images https://github.com/open-telemetry/opentelemetry-collector-releases

missed this comment. Sure - I'll create the issue over there also. thanks

**Comment by 0x6f677548** [comment]
Alternative steps to reproduce:
https://github.com/open-telemetry/opentelemetry-collector-releases/issues/662#issuecomment-2355473022

**Comment by ChrsMark** [comment]
@open-telemetry/collector-contrib-maintainers this looks like a valid suggestion and should be changed along with https://github.com/open-telemetry/opentelemetry-collector-releases/issues/662.

**Comment by rogercoll** [comment]
Fixing it in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36170

**Comment by 0x6f677548** [comment]
Hi @rogercoll, @jpkrohling 
this issue is still active. There is probably something wrong in the pipeline, as the dockerhub and ghcr images have not been updated with these changes.

Issue was marked as solved in 0.114.0 but images since then have the same issue. ex:  
https://hub.docker.com/layers/otel/opentelemetry-collector-contrib/0.114.0/images/sha256-43168fa5acb6989f40e8a2493275e30a62df8ee3f3dde1ba15c906cb8c9f3fb9

thanks

**Comment by rogercoll** [comment]
@0x6f677548 Although it might not be very useful because the binary is not packaged/distributed, the issue was fixed in this repository https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36170.

The container image you are linking, is published within the `opentelemetry-collector-releases` repository, there is an ongoing PR to make the changes there https://github.com/open-telemetry/opentelemetry-collector-releases/pull/738

**Comment by 0x6f677548** [comment]
@rogercoll , I see...
Sorry, I was not familiar with the way images were distributed. So, if I understood correctly, although this was added to this repo as fixed in [v0.114.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.114.0), it still needs to get merged to opentelemetry-collector-releases, is that it?

**Comment by codeboten** [comment]
This looks like it was resolved some time ago, closing
